services/cd-service/pkg/service/batch.go

```diff
@@ -147,6 +165,30 @@ func (d *BatchServer) processAction(
 			LockId:         act.LockId,
 			Authentication: repository.Authentication{RBACConfig: d.RBACConfig},
 		}, nil, nil
+	case *api.BatchAction_CreateEnvironmentTeamLock:
+		act := action.CreateEnvironmentTeamLock
+
+		if err := ValidateEnvironmentTeamLock("create", act.Environment, act.Team, act.LockId); err != nil {
+			return nil, nil, err
+		}
```

Comment:
this whole function is supposed to be deprecated. 
Please put the validations into the actual transformer.